File: studyhub-api/src/main/java/com/studyhub/studyhub_api/service/user_account/TeacherService.java
package com.studyhub.studyhub_api.service.user_account;

import com.studyhub.studyhub_api.dto.response.teacher.TeacherLiteResponse;

import java.util.List;

public interface TeacherService {
    List<TeacherLiteResponse> getTeacherList();
}

File: StudyHub_ag/src/app/features/guest/service/teacher/teacher.service.ts
import { Injectable } from "@angular/core";
import { BaseService } from "../../../../_service/base/base.service";
import { HttpClient } from "@angular/common/http";
import { API_ENDPOINTS } from "../../../../_service/base/api-endpoints";
import { Observable } from "rxjs";

@Injectable({
  providedIn: 'root',
})
export class TeacherService extends BaseService<TeacherLiteResponse>{
    constructor(http: HttpClient){
        super(http, API_ENDPOINTS.TEACHER)
    }

    getTeacherList(): Observable<ApiResponse<TeacherLiteResponse>>{
        return this.customRequest(
            "GET",
            API_ENDPOINTS.TEACHER_ENDPOINTS.LIST
        );
    }
}